File: src/saucer-base.hs
-- saucer base (jmcc)

import Sound.SC3.ID
import Sound.SC3.Lang.Control.OverlapTexture

saucer_base :: UGen
saucer_base =
    let a = rand 'α' 0 20
        b = rand 'α' 0 1000
        c = rand 'α' 0 5000
        p = rand 'α' (-1) 1
        o = sinOsc AR a 0 * b + (1.1 * b)
        o' = sinOsc AR o 0 * c + (1.1 * c)
    in pan2 (sinOsc AR o' 0 * 0.1) p 1

main :: IO ()
main = overlapTextureU (2,6,4,maxBound) saucer_base

File: gr/saucer-base.fs
\ SAUCER BASE (JMCC) #6

: R1 0 SWAP RAND.IR ;
: RLOC -1 1 RAND.IR 1 PAN2 ;
: OSC SINOSC.AR ;
: DRMS dup rot * swap 1.1 * + ;
: SB 20 R1 0 OSC 1000 R1 DRMS 0 OSC 5000 R1 DRMS 0 OSC 0.1 * RLOC ;
SB 2 6 4 INF OVERLAP-TEXTURE
